File: CS-Examples/02_Data/InsertHtmlStringIntoCell.vb
Imports Spire.Xls

Namespace InsertHtmlStringIntoCell
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Get the first sheet from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Define an HTML code string
            Dim htmlCode As String = "<div>first line<br>second line<br>third line</div>"

            ' Access the cell range A1 in the worksheet
            Dim range As CellRange = sheet("A1")

            ' Set the HTML string as the value of the cell range
            range.HtmlString = htmlCode

            ' Specify the filename for saving the workbook
            Dim result As String = "InsertHtmlStringIntoCell.xlsx"

            ' Save the workbook to the specified file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/02_Data/InsertHtmlStringIntoCell.cs
using Spire.Xls;
using System;
using System.Windows.Forms;

namespace InsertHtmlStringIntoCell
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Get the first sheet
            Worksheet sheet = workbook.Worksheets[0];

            // Inset html code to cell "A1"
            String htmlCode = "<div>first line<br>second line<br>third line</div>";
            CellRange range = sheet["A1"];
            range.HtmlString = htmlCode;

            // Specify the name for the resulting Excel file
            String result = "InsertHtmlStringIntoCell.xlsx";

            // Save the file
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to free up resources
            workbook.Dispose();

            // Launch the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
